cli/lakeflags/flags.go

```diff
@@ -95,3 +91,22 @@ func (l *Flags) URI() (*storage.URI, error) {
 	}
 	return u, err
 }
+
+// ClientURI returns the URI of the lake to connect to. If the lake path is
+// the defaultDataDir, it first checks if a zed service is running on
+// localhost:9867 and if so uses http://localhost:9867 as the lake location.
+func (l *Flags) ClientURI() (*storage.URI, error) {
+	u, err := l.URI()
+	if err != nil {
+		return nil, err
+	}
+	if u.String() == storage.MustParseURI(defaultDataDir).String() && localServer() {
+		u = storage.MustParseURI("http://localhost:9867")
+	}
```

Comment:
@mattnibs: This logic doesn't distinguish between not specifying a lake and specifying a lake at the same location as defaultDataDir. To make that more concrete, if `zed serve` is listening at localhost:9867 and I run `zed -lake $HOME/.local/share/zed ls`, it'll connect to `zed serve` even though I asked it to use $HOME/.local/share/zed. I don't think it should.

We could fix that by not assigning a default value to l.lake and then calling localServer and getDefaultLake only when l.lake isn't set.

Reply:
Good call, let's do that. Will fix.

Reply:
Actually that's not going to work since it's going to override it if someone does `zed ls -lake ''`.

Reply:
@nwt added back lackSpecified, ready for another review.